pkg/controller/operatorsource/catalogsourceconfigbuilder.go

```diff
@@ -37,8 +37,9 @@ func (b *CatalogSourceConfigBuilder) WithMeta(namespace, name string) *CatalogSo
 func (b *CatalogSourceConfigBuilder) WithOwner(owner *v1alpha1.OperatorSource) *CatalogSourceConfigBuilder {
 	trueVar := true
 	ownerReference := metav1.OwnerReference{
-		APIVersion: owner.APIVersion,
-		Kind:       owner.Kind,
+		APIVersion: fmt.Sprintf("%s/%s",
```

Comment:
We can't, because the owner we got from the SDK  doesn't have those fields populated anymore. Either we manually populate those fields in the owner and then use them as references, or we just modify the string here as I have done. Neither seems preferable, but at least in this case we aren't relying on the extent state of a value we didn't generate ourselves.

Reply:
Changing it in the builder is not the right spot. I would prefer manually populating them in the `owner` object using a helper  with a comment laying out the reasoning.

Reply:
Done